gef.py

```diff
@@ -3551,6 +3551,10 @@ def continue_handler(_: "gdb.ContinueEvent") -> None:
 
 def hook_stop_handler(_: "gdb.StopEvent") -> None:
     """GDB event handler for stop cases."""
+    if gef.session.remote:
```

Comment:
Wow ! How did I miss the cleanup here... Good catch!

Probably worth adding a quick test on this.

Reply:
This might be very slow syncing maps every time we stop (for single stepping) but i guess we need it.

Reply:
> This might be very slow syncing maps every time we stop (for single stepping) but i guess we need it.

I think to avoid syncing maps everytime we hit a breakpoint, we can wait for it to be used to pull it again, but I think the context rely a lot on this right ? So this would not change much things
But I don't think this would slow down too much (I didn't notice any differences while using it) since gdb does the exact same thing when he wants to check the proc maps (we are writing to tmpfs so there are no slow I/O i guess ?)

Reply:
Yeah it might be fine. We can run benchmarks against it to see how it changes.